src/main/java/org/opensearch/knn/quantization/enums/ScalarQuantizationType.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.enums;
+
+import lombok.Getter;
+
+/**
+ * The ScalarQuantizationType enum defines the various scalar quantization types that can be used
+ * for vector quantization. Each type corresponds to a different bit-width representation of the quantized values.
+ *
+ * <p>
+ * Future Developers: If you change the name of any enum constant, do not change its associated value.
+ * Serialization and deserialization depend on these values to maintain compatibility.
+ * </p>
```

Comment:
can we also add a test case which will break if any of the Quantization type has same Id?